lib/task-data/base-tasks/ucs-service-profile-discovery.js

```diff
@@ -0,0 +1,17 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+module.exports = {
+    friendlyName: 'UCS Service Profile Discovery Job',
+    injectableName: 'Task.Base.UCS.Service.Profile.Discovery',
+    runJob: 'Job.Ucs.Service.Profile.Discovery',
+    requiredOptions: [
+        'uri'
```

Comment:
From your implementation, the `username`, `password`, `ucs` are also required options.

```javascript
assert.string(this.options.uri);
assert.string(this.options.username);
assert.string(this.options.password);
assert.string(this.options.ucs);
```